src/markdown_exec/_internal/mkdocs_plugin.py

```diff
@@ -52,18 +54,29 @@ def _get_logger(name: str) -> _LoggerAdapter:
 class MarkdownExecPluginConfig(Config):
     """Configuration of the plugin (for `mkdocs.yml`)."""
 
-    ansi = config_options.Choice(("auto", "off", "required", True, False), default="auto")
+    ansi = config_options.Choice(
+        ("auto", "off", "required", True, False), default="auto"
+    )
     """Whether the `ansi` extra is required when installing the package."""
     languages = config_options.ListOfItems(
         config_options.Choice(formatters.keys()),
         default=list(formatters.keys()),
     )
     """Which languages to enabled the extension for."""
+    auto_exec = config_options.Type(
+        (list, str),
+        default=None,
+    )
+    """Languages for which to automatically execute code blocks."""
 
 
 class MarkdownExecPlugin(BasePlugin[MarkdownExecPluginConfig]):
     """MkDocs plugin to easily enable custom fences for code blocks execution."""
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.original_env_vars = {}
+
```

Comment:
Is there a simpler way to do this? can we define this in "on_config"? or would htat introduce additional complication